pkg/datagatherer/versionchecker/versionchecker_test.go

```diff
@@ -0,0 +1,357 @@
+	if err != nil {
+		t.Fatalf("failed to marshal data %s", err)
+	}
+
+	expectedResultsJSON := fmt.Sprintf(`[
+  {
+    "pod": {
+      "kind": "Pod",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "example-6d94489854-zpzhr",
+        "namespace": "example",
+        "selfLink": "/api/v1/namespaces/example/pods/example-6d94489854-zpzhr",
+        "uid": "efff9dae-28ca-42c3-be70-970731c44f67",
+        "resourceVersion": "32023849",
+        "creationTimestamp": null,
+        "labels": {
+          "app": "example"
+        },
+        "ownerReferences": [
+          {
+            "apiVersion": "apps/v1",
+            "kind": "ReplicaSet",
+            "name": "example-6d94489854",
+            "uid": "bb6c0f31-0e28-4c28-a81d-91b8d7bfed33",
+            "controller": true,
+            "blockOwnerDeletion": true
+          }
+        ]
+      },
+      "spec": {
+        "containers": [
+          {
+            "name": "example",
+            "image": "%s/jetstack/example:v1.0.0",
+            "command": [
+              "sh",
+              "-c"
+            ],
+            "resources": {}
+          }
+        ]
+      },
+      "status": {
+        "containerStatuses": [
+          {
+            "name": "example",
+            "state": {},
+            "lastState": {},
+            "ready": false,
+            "restartCount": 0,
+            "image": "",
+            "imageID": "is set"
+          }
+        ]
+      }
+    },
+    "result": {
+      "CurrentVersion": "v1.0.0",
+      "LatestVersion": "v1.0.1",
+      "IsLatest": false,
+      "ImageURL": "%s/jetstack/example"
+    }
+  }
+]`, parsedURL.Host, parsedURL.Host)
+
+	if expectedResultsJSON != string(resultsJSON) {
+		t.Fatalf("results json does not match: %s vs %s", resultsJSON, expectedResultsJSON)
+	}
+}
+
+// config must be loaded from file paths, this creates a tmp file with the host
+// to load in for the DataGatherer
+func createDgHostConfigWithServer(server string) (*os.File, error) {
+	tmpfile, err := ioutil.TempFile("", "example")
+	if err != nil {
+		return nil, fmt.Errorf("failed to create a tmpfile for host")
+	}
+
+	if _, err := tmpfile.Write([]byte(server)); err != nil {
+		return nil, fmt.Errorf("failed to write to tmp host file")
+	}
+	if err := tmpfile.Close(); err != nil {
+		return nil, fmt.Errorf("failed to close tmp host file after writing")
+	}
+
+	return tmpfile, nil
+}
+
+// creates a kubeconfig file on disk with a reference to the local server
+// allowing requests to be mocked
+func createKubeConfigWithServer(server string) (*os.File, error) {
+	content := fmt.Sprintf(`
+apiVersion: v1
+kind: Config
+clusters:
+- cluster:
+    server: %s
+  name: example
+contexts:
+- context:
+    cluster: example
+    namespace: default
+    user: test
+  name: test
+current-context: test
+users:
+- name: test
+  user:
+    username: test
+    password: test`, server)
+	tmpfile, err := ioutil.TempFile("", "example")
```

Comment:
Perhaps we create a testing directory for this testing suite, and then remove the directory at the end of the test?

The creation of the dir, and deletion can be done in the test main, and then we don't have to remember cleaning up per test.

Reply:
There are only two files written during the test at the moment so this is acceptable to me. If we add more, we can come up with a better way to do it.

I made the prefixes more descriptive in 1508c19592d2847d97559c2dcd610515be6f3738.